rfc/001-identifier-triplets.md

```diff
@@ -0,0 +1,201 @@
+store in the blocking table.  Currently, that is limited to 20 characters which is a bit arbitrary,
+but the idea is to keep the values small so we can index them efficiently.  If we keep that limit, we
+need to use identifier values that are guaranteed to be less than that limit.
+
+Previous research has shown that the last **4 characters of an identifier value** are often the most
+unique and can be used to block on.  However, we should likely also include elements of the type and
+authority, if we want to provide some assurance that we are not blocking on a different identifier
+all together. For that, we recommend storing the **entire type** (which is limited to 7 characters)
+and the **first 2 characters of the authority** (which is free-form).
+
+| patient_id | blocking_key | value      |
+|------------|--------------|------------|
+| 1          | IDENTIFIER   | DL:CA:3456 |
+| 1          | IDENTIFIER   | SS::6789   |
+
+### Evaluation
+
+As indicated in the business rules section, evaluation on identifiers can happen in two ways. The
+first way is to evaluate on any identifier, meaning that the evaluation step will result in a
+match if any two identifiers between the documents are a match.  The second way is to evaluate on
+a specific type of identifier, meaning that the evaluation step will result in a match if the 
+specified type of identifier between the documents is a match.
+
+The evaluation functions will be comparing all 3 parts of the identifier tuple (type, authority,
+value) when determining if two identifiers are a match.  The difference is just between what types
+of identifiers are we going to compare.
+
+```python
+
+if feature == 'IDENTIFIER':
+    assert values == ['DL:CA:A123456', 'SS::123-45-6789']
+if feature == 'IDENTIFIER:SS':
+    assert values == ['SS::123-45-6789']
+```
+
+## Alternatives Considered
+
+The main alternative to a generic identifier triplet is to continue creating specific fields (eg 
+`MRN`, `SSN`, `drivers_license`) for each identifier type.  This would require more configuration
+and would not be as flexible as the proposed solution. However, this does allow for customization's
+when blocking or evaluating on specific identifier types.  For example, we know that SSN will never
+require an authority, so we can make a slight reduction in the blocking value size knowing that is
+never required.  In the case of Driver's License, we know that the authority will always be a state,
+so we could implement custom normalization logic attempting to standardize the state values (eg `CA`
+vs `California`).  This is more work long-term, and likely more confusing for users as each identifier
+field has slightly different behavior, but does allow for maximum flexibility.
+
+## Risks and Drawbacks
+
+- Variations in authority values could lead to false negatives in comparisons (eg `CA` vs `California`).
+- Variations in value formats could lead to false negatives in comparisons (eg `123-45-6789` vs `123456789`).
```

Comment:
That's a good question, I think we could safely strip leading and trailing whitespace.   I think we might be able to get away with stripping all whitespace characters from the value as well.  I don't think we can safely remove any other values though, as they might have significance to particular types.  There are just too many specified here for me to know the implications it would have on all of them https://terminology.hl7.org/6.0.2/CodeSystem-v2-0203.html.

Reply:
Sounds good! Sounds like stripping whitespace is sufficient.